Simply overwrite an existing installation without asking questions

diff -ru a/Makefile b/Makefile
--- a/Makefile	2017-11-21 13:35:13.000000000 +0100
+++ b/Makefile	2018-06-29 19:48:19.654035809 +0200
@@ -22,9 +22,6 @@
 	@echo ">>> WARNING  CoCoALib installation procedure is still experimental! <<<"
 	@echo "======================================================================="
 	@echo
-	@echo "Continuing with installation after 5 secs..."
-	@echo
-	@sleep 5
 	@if [ \! -f lib/libcocoa.a ]; \
 	 then \
 	   echo "***** INSTALLATION ERROR: CoCoALib is not built!           *****"; \
