diff --git a/Makefile b/Makefile
index 1d5fad9..29fb0f1 100644
--- a/Makefile
+++ b/Makefile
@@ -138,13 +138,12 @@ MKDIR=mkdir -p

 # PREFIX = /usr/local/gcc/6.2.0/bin/
 # PREFIX = /usr/local/gcc-8.1/bin/
-PREFIX =
 SHELL       = /bin/sh
 #ARCH        = LINUX

-CC          = $(PREFIX)gcc
+#CC          = $(PREFIX)gcc-15
 CLINKER     = $(CC)
-CXX         = $(PREFIX)g++
+#CXX         = $(PREFIX)g++-15
-CCLINKER    = $(CXX)
+#CCLINKER    = $(CXX)

 #CC          = $(PREFIX)gcc-8.1
diff --git a/src/field.h b/src/field.h
index 866be22..bc50a12 100644
--- a/src/field.h
+++ b/src/field.h
@@ -208,6 +208,7 @@ class FieldElementImplementation
   {
 	  fprintf(stderr,"*this is not in Z/pZ.\n");
 	  assert(0);
+	  return 0;
   }
   virtual bool isInteger()const
   {
@@ -223,6 +224,7 @@ class FieldElementImplementation
   Field& operator=(const Field& a)
     {
       assert(0);
+      return const_cast<Field&>(a);
     }//assignment
 };

@@ -271,6 +273,7 @@ class FieldImplementation
   virtual FieldElement random()
   {
 	  assert(0);
+	  return 0;
   }
     virtual int getCharacteristic()const=0;
     virtual const char *name()=0;
diff --git a/src/gfanlib_circuittableint.h b/src/gfanlib_circuittableint.h
index 6078a36..fdf4a36 100644
--- a/src/gfanlib_circuittableint.h
+++ b/src/gfanlib_circuittableint.h
@@ -27,7 +27,7 @@ namespace gfan{
   template<typename> struct MyMakeUnsigned;
   template <> struct MyMakeUnsigned<int32_t>{typedef uint32_t type;};
   template <> struct MyMakeUnsigned<int64_t>{typedef uint64_t type;};
-  template <> struct MyMakeUnsigned<__int128>{typedef unsigned __int128 type;};
+  template <> struct MyMakeUnsigned<__int128_t>{typedef __uint128_t type;};

   class MVMachineIntegerOverflow: public std::exception
 {
diff --git a/src/gfanlib_z.h b/src/gfanlib_z.h
index 2908d25..7e8c66f 100644
--- a/src/gfanlib_z.h
+++ b/src/gfanlib_z.h
@@ -9,6 +9,7 @@
 #define LIB_Z_H_

 #include <string.h>
+#include <cstdint>
 #include <ostream>
 #include <iostream>
 #include <cstdint>
--
2.53.0
