diff --git a/Makefile b/Makefile
index 9556cf4..40127b6 100644
--- a/Makefile
+++ b/Makefile
@@ -22,20 +22,8 @@ STACTDUMP_OPTIONS = -DSTACKDUMP_ENABLED -D__assert_fail=__assert_fail2
 endif
 
 
-ifeq ($(sagepath),)
-SAGE_LINKOPTIONS = 
-SAGE_INCLUDEOPTIONS =
-else
-SAGE_LINKOPTIONS = -L$(sagepath)/ -lpython2.6 -lcsage -lsingular
-SAGE_INCLUDEOPTIONS = -I $(sagepath)/
-SAGE_OBJECTS = sage.o sage_link.so
-
-sage_link.so: sage_link.pyx setup.py
-	python setup.py build_ext --inplace --pyrex-include-dirs=$(SAGE_ROOT)/devel/sage/
-endif
-
 ifeq ($(gmppath),)
-GMP_LINKOPTIONS = -lgmp
+GMP_LINKOPTIONS = -lgmp -lm
 GMP_INCLUDEOPTIONS =
 else
 GMP_LINKOPTIONS = $(gmppath)/lib/libgmp.a
@@ -49,7 +37,7 @@ CDDDEFINE_PREFIX = -DNOCDDPREFIX
 endif
 
 ifeq ($(cddpath),)
-CDD_LINKOPTIONS = -L/usr/local -lcddgmp
+CDD_LINKOPTIONS = -lcddgmp
 CDD_INCLUDEOPTIONS =
 else
 CDD_LINKOPTIONS = $(cddpath)/lib/libcddgmp.a
@@ -125,11 +82,10 @@ OPTFLAGS    =  -DGMPRATIONAL -Wuninitialized -fno-omit-frame-pointer -O2	 #-O3 -
 #OPTFLAGS    =  -DGMPRATIONAL -Wuninitialized -fno-omit-frame-pointer -O3 -msse2 -ftree-vectorizer-verbose=2 -ffast-math #-DNDEBUG
 #OPTFLAGS    =  -DGMPRATIONAL -Wuninitialized -fno-omit-frame-pointer -O3 -mavx -msse2 -ftree-vectorizer-verbose=2 -ffast-math #-DNDEBUG
 
-CFLAGS	  = $(OPTFLAGS) $(GPROFFLAG) $(STACTDUMP_OPTIONS) $(ADDITIONALINCLUDEOPTIONS) -g $(CDDDEFINE_PREFIX) -std=c++20 $(TBB_INCLUDEOPTIONS) #-pedantic
-#CFLAGS	  = $(OPTFLAGS) $(GPROFFLAG) $(STACTDUMP_OPTIONS) $(ADDITIONALINCLUDEOPTIONS) -std=c++0x -g $(CDDDEFINE_PREFIX) #-pedantic
-#CFLAGS	  = $(OPTFLAGS) $(GPROFFLAG) $(STACTDUMP_OPTIONS) $(ADDITIONALINCLUDEOPTIONS) -D_GLIBCXX_DEBUG -g $(CDDDEFINE_PREFIX) #-pedantic
-CCFLAGS	  = $(CFLAGS)
-FFLAGS	  = $(OPTFLAGS)
+CFLAGS	  := $(CFLAGS) $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
+CXXFLAGS  := $(CXXFLAGS) $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) -std=c++20 $(TBB_INCLUDEOPTIONS) #-pedantic
+
+SINGULAR_OBJECTS = src/polynomialgcd.o
 
 CATSOBJECTS =	lp_cdd.o \
 		parser.o \
@@ -394,15 +351,14 @@ EXECS	  = $(MAIN)
 # (compiling with gcc version 4.7.2 and running gfan _tropicaltraverse on a starting cone for Grassmann3_7)
 # Either this is a bug in the code or in the compiler. The bug disappears by compiling with -fno-guess-branch-probability
 # SYMMETRICTRAVERSAL_FLAGS is set to -fno-guess-branch-probability on Linux/gcc and empty on macOS/clang.
 src/symmetrictraversal.o: src/symmetrictraversal.cpp
-	$(CXX) $(CFLAGS) $(SYMMETRICTRAVERSAL_FLAGS) -c src/symmetrictraversal.cpp -o src/symmetrictraversal.o
+	$(CXX) $(CXXFLAGS) $(SYMMETRICTRAVERSAL_FLAGS) -c src/symmetrictraversal.cpp -o src/symmetrictraversal.o
 
 # Define suffixes to make the program compile on legolas.imf.au.dk :
 .SUFFIXES: .o .cpp .c
 
 OBJECTS1 = 	$(addprefix src/,$(SOPLEX_OBJECTS)) \
 		$(SINGULAR_OBJECTS) \
-		$(SAGE_OBJECTS) \
 		$(addprefix src/,$(CATSOBJECTS)) \
 		$(addprefix src/,$(APPOBJECTS)) \
 		src/symmetrictraversal.o
@@ -420,6 +376,6 @@ default: $(OBJECTS) $(ADDITIONALOBJECTS) $(EXECS)
 
 $(MAIN): $(OBJECTS)
-	$(CCLINKER) $(OBJECTS) $(ADDITIONALLINKOPTIONS) $(GPROFFLAG) -lpthread $(PLATFORM_LINKOPTIONS) -o $(MAIN)
+	$(CXX) $(CXXFLAGS) -pthread -o $(MAIN) $(LDFLAGS) $(OBJECTS) -lpthread $(ADDITIONALLINKOPTIONS) $(GPROFFLAG) $(PLATFORM_LINKOPTIONS)
 
 release:
 	rm -f -r $(RELEASEDIR)/*
@@ -546,7 +502,7 @@ check:
 .c.o:
 	$(CC) $(CFLAGS) -c $< -o $(patsubst %.c,%.o,$<)
 .cpp.o:
-	$(CXX) $(CFLAGS) -c $< -o $(patsubst %.cpp,%.o,$<)
+	$(CXX) $(CXXFLAGS) -c $< -o $(patsubst %.cpp,%.o,$<)
 
 # wget http://ftp.sunet.se/pub/gnu/gmp/gmp-4.2.2.tar.gz
 # tar -xzvf gmp-4.2.2.tar.gz
